lua/bufferline/render.lua

```diff
@@ -82,11 +82,13 @@ local function render()
   end
 
   local opts = vim.g.bufferline
-  local icons = vim.g.icons
+
+  -- Ensure that the new setting scheme is adhered to.
+  local icons = type(opts.icons) == 'table' and opts.icons or vim.fn['bufferline#init_icons']()
```

Comment:
Being a `table` indicates that the new, proposed `g:bufferline.icons` scheme is being used. If `opts.icons` is not a table, then the initialization step for icons should be run again.